src/Dependencies/Collections/SegmentedArray.cs

```diff
@@ -13,13 +13,57 @@ internal static class SegmentedArray
         /// <seealso cref="Array.Clear(Array, int, int)"/>
         internal static void Clear<T>(SegmentedArray<T> buckets, int index, int length)
         {
-            throw new NotImplementedException();
+            if (index < 0 || length < 0 || (uint)(index + length) > (uint)buckets.Length)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
```

Comment:
Consider breaking this up and including the argument that was out of range.

Reply:
This is the same check used for `Array.Clear`:

https://github.com/dotnet/runtime/blob/5078dd61a81c88dbdd131a77b8148b4dee3803fa/src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs#L290-L291

Reply:
With that said, this class is _weird_ - the instinct is to keep all the original coding approach from `Array` but it's both ugly and difficult to port directly. So I could really go either way on it.